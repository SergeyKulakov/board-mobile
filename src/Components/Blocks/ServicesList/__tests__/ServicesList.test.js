import React from 'react'
import { shallow } from 'enzyme'

import 'jest-styled-components'

import Component from '../ServicesList'

describe('ServicesList', () => {
  const defaultProps = {
    horizontal: false,
    data: [],
    withShadowBox: false,
    orientation: 'LANDSCAPE',
    changedItemId: undefined,
    numColumns: undefined,
    activeItemId: undefined,
    disabledItems: undefined,
    activeItems: undefined,
    title: 'title',
    isSelectedAll: false,
    onClickItem: () => null,
    getServiceTitle: text => text,
    onSelectAll: () => null,
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    const withShadowBoxWrapper = shallow(
      <Component {...defaultProps} withShadowBox />,
    )

    const args = {
      item: {
        categoryId: '23',
        _id: '123',
        isActive: false,
        title: 'title',
        iconName: 'iconName',
      },
    }

    const renderItemWrapper = wrapper
      .find('FlatList')
      .props()
      .renderItem(args)

    expect([wrapper, withShadowBoxWrapper, renderItemWrapper]).toMatchSnapshot()
  })
})
