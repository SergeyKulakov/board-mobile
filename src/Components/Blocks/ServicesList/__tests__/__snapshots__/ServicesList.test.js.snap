// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesList render snapshot 1`] = `
Array [
  <Styled(View)
    mt={false}
  >
    <FlatList
      data={
        Array [
          Object {
            "_id": "all",
            "iconName": "All.png",
            "isActive": false,
            "isChanged": false,
            "onClick": [Function],
            "title": "All",
          },
        ]
      }
      disableVirtualization={false}
      extraData={
        Array [
          "",
          0,
          undefined,
        ]
      }
      horizontal={false}
      initialNumToRender={10}
      key="v"
      keyExtractor={[Function]}
      listKey="ServicesList"
      maxToRenderPerBatch={10}
      numColumns={5}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Styled(View)>,
  <ShadowBox
    style={
      Object {
        "maxWidth": "auto",
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
  >
    <Component
      style={
        Object {
          "paddingLeft": "5%",
          "paddingRight": "5%",
        }
      }
    >
      T
      itle
    </Component>
    <Styled(View)
      mt={20}
    >
      <FlatList
        data={
          Array [
            Object {
              "_id": "all",
              "iconName": "All.png",
              "isActive": false,
              "isChanged": false,
              "onClick": [Function],
              "title": "All",
            },
          ]
        }
        disableVirtualization={false}
        extraData={
          Array [
            "",
            0,
            undefined,
          ]
        }
        horizontal={false}
        initialNumToRender={10}
        key="v"
        keyExtractor={[Function]}
        listKey="ServicesList"
        maxToRenderPerBatch={10}
        numColumns={5}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </Styled(View)>
  </ShadowBox>,
  <ForwardRef>
    <ListItem
      getText={[Function]}
      iconName="iconName"
      isActive={false}
      isChanged={false}
      isShowFolderIcon={false}
      onClick={[Function]}
      title="title"
    />
  </ForwardRef>,
]
`;
